include/bitcoin/bitcoin/chain/block.hpp

```diff
@@ -157,19 +158,20 @@ class BC_API block
     bool is_extra_coinbases() const;
     bool is_final(size_t height, uint32_t block_time) const;
     bool is_distinct_transaction_set() const;
-    bool is_valid_coinbase_claim(size_t height) const;
+    bool is_valid_coinbase_claim(size_t height, const settings& settings) const;
     bool is_valid_coinbase_script(size_t height) const;
     bool is_valid_witness_commitment() const;
     bool is_forward_reference() const;
     bool is_internal_double_spend() const;
     bool is_valid_merkle_root() const;
     bool is_segregated() const;
 
-    code check(bool retarget) const;
-    code check_transactions() const;
-    code accept(bool transactions=true, bool header=true) const;
-    code accept(const chain_state& state, bool transactions=true,
+    code check(bool retarget, const settings& settings) const;
```

Comment:
For consistency (see tx) the `retarget` parameter should be second and default to `true`

Reply:
makes sense